src/workflows-app/components/SubmitWorkflowModal.test.ts

```diff
@@ -102,284 +97,170 @@ const postRunSetPayload = expect.objectContaining({
-  });
-
-  it('should allow submit to workflows app by writer without a cromwell-runner', async () => {
-    // ** ARRANGE **
-    const user = userEvent.setup();
-    const postRunSetFunction = jest.fn();
-    const createAppV2 = jest.fn();
-    const listAppsV2 = jest
-      .fn()
-      .mockImplementationOnce(() => Promise.resolve([]))
-      .mockImplementation(() => Promise.resolve(mockCollaborativeAzureApps));
-    asMockedFn(getTerraUser).mockReturnValue({ email: 'groot@gmail.com' });
-
-    await asMockedFn<() => DeepPartial<AjaxContract>>(Ajax).mockImplementation(() => {
-      return {
-        Apps: {
-          createAppV2,
-          listAppsV2,
-        },
-        Cbas: {
-          runSets: {
-            post: postRunSetFunction,
+describe('SubmitWorkflowModal', () => {
+  const testCases: Array<{
+    featureFlagEnabled: boolean;
+    role: 'CREATOR' | 'WRITER' | 'READER';
+    canSubmit: boolean;
+    cromwellRunnerStates: Array<'NONE' | 'RUNNING' | 'PROVISIONING'>;
+  }> = [
+    {
+      featureFlagEnabled: false,
+      role: 'CREATOR',
+      canSubmit: true,
+      cromwellRunnerStates: [],
+    },
+    {
+      featureFlagEnabled: false,
+      role: 'WRITER',
+      canSubmit: false,
+      cromwellRunnerStates: [],
+    },
+    {
+      featureFlagEnabled: false,
+      role: 'READER',
+      canSubmit: false,
+      cromwellRunnerStates: [],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'CREATOR',
+      canSubmit: true,
+      cromwellRunnerStates: ['RUNNING'],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'WRITER',
+      canSubmit: true,
+      cromwellRunnerStates: ['RUNNING'],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'READER',
+      canSubmit: false,
+      cromwellRunnerStates: [],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'CREATOR',
+      canSubmit: true,
+      cromwellRunnerStates: ['PROVISIONING', 'RUNNING'],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'WRITER',
+      canSubmit: true,
+      cromwellRunnerStates: ['PROVISIONING', 'RUNNING'],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'READER',
+      canSubmit: false,
+      cromwellRunnerStates: [],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'CREATOR',
+      canSubmit: true,
+      cromwellRunnerStates: ['NONE', 'PROVISIONING', 'RUNNING'],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'WRITER',
+      canSubmit: true,
+      cromwellRunnerStates: ['NONE', 'PROVISIONING', 'PROVISIONING', 'RUNNING'],
+    },
+    {
+      featureFlagEnabled: true,
+      role: 'READER',
+      canSubmit: false,
+      cromwellRunnerStates: [],
+    },
+  ];
+
+  it.each(testCases)(
+    'should be able to submit with role=$role, feature flag enabled=$featureFlagEnabled, cromwell runner status=$cromwellRunnerStates: $canSubmit',
```

Comment:
Tiny nit, might be nicer to put the `$canSubmit` change right at the front (eg "should be able to submit [...]" or "should **not** be able to submit [...]"